scripts/generate-shaders.js

```diff
@@ -82,18 +81,21 @@ ${fragmentPrelude}
                     "{name} = unpack_mix_{a_type}(a_{name}, a_{name}_t);"
                 ]
             });
-    }
 
-    function fragmentSource() {
-        const source = fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8');
-        return applyPragmas(source, {
+    const fragmentSource = applyPragmas(fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8'), {
                 define: [
                     "varying {precision} {type} {name};"
                 ],
                 initialize: [
                 ]
             });
-    }
+
+    const shasum = crypto.createHash('sha1');
```

Comment:
Another approach would be to use a unique version per build, e.g. the commit hash, along with the name of the shader.